### PR TITLE
nat: enable upper case proto

### PR DESCRIPTION
**Pull request by chenhanxiao**
We only accepted lower case proto:
tcp, udp.
This patch will enable us to use upper case
of proto such as: EXPOSE 1234/TCP

Signed-off-by: Chen Hanxiao chenhanxiao@cn.fujitsu.com

### CODE REVIEW

**Comment by jfrazelle** [comment]
LGTM

**Comment by jfrazelle** [comment]
I would be nice to have a test added to the unit tests

**Comment by LK4D4** [comment]
+1 to test

**Comment by tiborvass** [comment]
@chenhanxiao would you mind adding a unit test please? Let us know if you need any help

**Comment by chenhanxiao** [comment]
@tiborvass @jfrazelle 

Sure, I'll add a test case for it.

PS: for the sake of my GMT+8, I may not reply the comments so quikly

**Comment by chenhanxiao** [comment]
@tiborvass 
how could I just run a specific testcases?
ex:
integration-cli/docker_cli_build_test.go

Thanks

**Comment by LK4D4** [comment]
@chenhanxiao 
```
TESTFLAGS="-run TestBuild" make test-integration-cli
```
will run all tests which names starting from `TestBuild`

**Comment by chenhanxiao** [comment]
@LK4D4 
Thanks for you help.
I'll try and add a test cases to my PR.

**Comment by chenhanxiao** [comment]
@jfrazelle  @LK4D4 @tiborvass 

testcase added, please check.

**Comment by cpuguy83** [comment]
LGTM

File: integration-cli/docker_cli_build_test.go [hunk 1]
```diff
@@ -2293,6 +2293,27 @@ func TestBuildExposeOrder(t *testing.T) {
 	logDone("build - expose order")
 }
 
+func TestBuildExposeUpperCaseProto(t *testing.T) {
+	name := "testbuildexpose"
```
Comment (tiborvass) [inline integration-cli/docker_cli_build_test.go]: @chenhanxiao `testbuildexposeuppercaseproto`

Comment (chenhanxiao) [inline integration-cli/docker_cli_build_test.go]: @tiborvass 
changed, please check.

BTW, I'll be on the trip soon.
If there are other comments, I had to do it in Thuesday or later.

Thanks

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**